eds/scripts/personalization.js

```diff
@@ -15,14 +15,152 @@ import {
 } from './personalizationUtils.js';
 import {DX_PROGRAM_TYPE} from "../blocks/utils/dxConstants.js";
 
+async function replaceProfileImage(elements) {
+  try {
+    const accessToken = window.adobeIMS.getAccessToken();
+    if (!accessToken?.token) {
+      elements.forEach(el => el.remove());
+      return;
+    }
+
+    const miloLibs = getLibs();
+    const { getConfig } = await import(`${miloLibs}/utils/utils.js`);
+
+    const { env } = getConfig();
+    const headers = new Headers({ Authorization: `Bearer ${accessToken.token}` });
+    const profileResponse = await fetch(`https://${env.adobeIO}/profile`, { headers });
+
+    if (profileResponse.status !== 200) {
+      elements.forEach(el => el.remove());
+      return;
+    }
+
+    const profileData = await profileResponse.json();
+
+    if (!profileData?.user?.avatar) {
+      elements.forEach(el => el.remove());
+      return;
+    }
+
+    const userAvatar = profileData.user.avatar;
+
+    elements.forEach((el) => {
+      const textNode = Array.from(el.childNodes).find(
+        node => node.nodeType === Node.TEXT_NODE && node.textContent.includes('$profileImage')
+      );
+
+      if (!textNode) return;
+
+      const img = document.createElement('img');
+      img.src = userAvatar;
+      img.alt = '';
+      img.dataset.userProfile = 'true';
+      img.width = 96;
+      img.height = 96;
+
+      const picture = document.createElement('picture');
+      picture.appendChild(img);
+
+      if (el.tagName === 'P') {
+        el.classList.add('icon-area');
+      }
+
+      el.replaceChild(picture, textNode);
+    });
+  } catch (error) {
+    console.warn('Failed to replace profile image placeholders:', error);
+    elements.forEach(el => el.remove());
+  }
+}
+
+function personalizeProfileImage(elements) {
+  if (!elements.length) return;
+
+  window.addEventListener('dxp:imsReady',  () => {
+     replaceProfileImage(elements);
+  });
+}
+
+async function replaceCompanyLogo(elements) {
+  try {
+    const accessToken = window.adobeIMS.getAccessToken();
```

Comment:
it's crazy to do a thing like this only for a profile image. This can't be the way to go, but also a cookie is not a good option for a whole URL...